src/utils/fetchData.ts

```diff
@@ -0,0 +1,37 @@
+import { Platform } from 'react-native';
+
+export function fetchText(uri?: string): Promise<string | null> {
+  if (
+    uri &&
```

Comment:
Could we make a guard here? If both functions `dataUriToXml` and `fetchUriData` expect string, then we can add this here
```suggestion
  if (!uri) {
    return null;
  }
```
and then remove the check from `fetchUriData`

Reply:
Good point, thanks.

Reply:
done